src/lp_sensitivity.jl

```diff
@@ -0,0 +1,322 @@
+    basic = var_basis_status .== Ref(MOI.BASIC)
+    B = A[:, basic]
+    x_B = x[basic]
+    local rho::Vector{Float64}
+    if F <: MOI.SingleVariable
+        var = constraint_object(constraint).func
+        j = findfirst(isequal(var), all_variables(model))
+        rho = -B \ Vector(A[:, j])
+    else
+        i = findfirst(isequal(constraint), constraints)
+        ei = zeros(size(A, 1))
+        ei[i] = 1.0
+        rho = B \ ei
+    end
+    UmX_B = U[basic] - x_B
+    LmX_B = L[basic] - x_B
+    pos = rho .> 1e-7
+    neg = rho .< -1e-7
+    return maximum([-Inf; UmX_B[neg] ./ rho[neg]; LmX_B[pos] ./ rho[pos]]),
+            minimum([Inf; UmX_B[pos] ./ rho[pos]; LmX_B[neg] ./ rho[neg]])
+end
+
+"""
+Returns the optimal reduced costs for the variables of the problem in standard form, c.f. `_std_matrix(model)`
+"""
+function _std_reduced_costs(model::Model, constraints::Vector{ConstraintRef})
+    con_types = list_of_constraint_types(model)
+    vars = all_variables(model)
+    var_to_idx = Dict{VariableRef, Int64}()
+    for (j, var) in enumerate(vars)
+        var_to_idx[var] = j
+    end
+    y = [zeros(length(vars)); dual.(constraints)]
+
+    for (F, S) in con_types
+        if F <: VariableRef
+            constraints = all_constraints(model, F, S)
+            for constraint in constraints
+                con_obj = constraint_object(constraint)
+                j = var_to_idx[con_obj.func]
+                y[j] += dual(constraint)
+            end
+            #issue 1892 makes is hard to know how to handle variable constraints, atm. assume unique variable constraints
+        end
+    end
+
+    if objective_sense(model) == MOI.MAX_SENSE
+        y .*= -1.0
+    end
+    return y
+end
+
+"""
+    lp_objective_perturbation_range(var::VariableRef)::Tuple{Float64, Float64}
+
+Gives the range by which the cost coefficient can change and the current lp-basis
+remains optimal, i.e., the reduced costs remain valid.
+
+## Notes
+- The range denote valid changes, Δ in [l, u], for which c[var] += Δ do not violate the current optimality conditions.
+
+"""
+function lp_objective_perturbation_range(var::VariableRef)::Tuple{Float64, Float64}
+    model = owner_model(var)
+    if termination_status(model) != MOI.OPTIMAL
+        error("The perturbation range of the objective is not available because the current solution "*
+              "is not optimal.")
+    end
+    if objective_sense(model) == MOI.FEASIBILITY_SENSE
+        error("The perturbation range of the objective is not applicable on feasibility problems.")
+    end
+    if !has_duals(model)
+        error("The perturbation range of the objective is not available because no dual result is " *
+              "available.")
+    end
+    # Optimization possible: since the has_upper_bound(var) and has_lower_bound(var)
+    # Do not capture all variable bounds (issue 1892), it's hard to directly
+    # access the reduced cost of a variable, now the complete matrix is always built.
```

Comment:
I don't understand this comment. Doesn't `_std_reduced_costs` directly access the reduced costs of the variables?